### PR TITLE
Warn and log

### PR DESCRIPTION
**Pull request by sjpfenninger**
Fixes issue(s) #226

Summary of changes in this pull request:

* Revised logging and warnings

Reviewer checklist:

- [x] Test(s) added to cover contribution
- [x] Documentation updated
- [x] Changelog updated
- [x] Coverage maintained or improved

### CODE REVIEW

File: calliope/test/test_core_util.py [hunk 1]
```diff
@@ -166,32 +166,19 @@ def test_memoize_instancemethod(self):
 
 
 class TestLogging:
```
Comment (brynpickering) [inline calliope/test/test_core_util.py]: You can easily test this with logging.getLogger('calliope').getEffectiveLevel(), whereby critical = 50 down to debug = 10. So i'd reinstate this test ;)

File: calliope/core/util/logging.py [hunk 2]
```diff
@@ -10,53 +10,40 @@
...
-    Level options (in descending order of verbosity):
-
-    * 'DEBUG'
-    * 'SOLVER' -> Calliope custom level, assigned value of 19,
-                  returns solver (e.g. GLPK) stream
-    * 'INFO'
-    * 'WARNING' -> default level
-    * 'ERROR'
-    * 'CRITICAL'
+def set_log_verbosity(verbosity='info', include_solver_output=True, logger=None):
```
[the remark this answers is not in the document]
Comment (sjpfenninger) [reply calliope/core/util/logging.py]: Fixed

File: calliope/core/util/logging.py [hunk 3]
```diff
@@ -10,53 +10,40 @@
...
+        backend_logger.setLevel(logging.DEBUG)
+    if logger is None:
+        logger = logging.getLogger()  # Root logger
+        logger.propagate = False
+        console = logging.StreamHandler(stream=sys.stdout)
+        logger.addHandler(console)
+    logger.setLevel(verbosity.upper())
+
+
+def log_time(logger, timings, identifier, comment=None, level='info', time_since_start=False):
```
[the remark this answers is not in the document]
Comment (sjpfenninger) [reply calliope/core/util/logging.py]: Fixed

File: calliope/test/test_core_util.py (hunk 1 above)
Comment (sjpfenninger) [reply calliope/test/test_core_util.py]: Fixed

**Pull request merged**